dbms/src/Interpreters/Join.cpp

```diff
@@ -127,15 +123,27 @@ Join::Join(
             strictness = ASTTableJoin::Strictness::All;
         }
     }
+    if (!left_filter_column.empty() && !isLeftJoin(kind))
+        throw Exception("Not supported: non left join with left conditions");
+    if (!right_filter_column.empty() && !isRightJoin(kind))
+        throw Exception("Not supported: non right join with right conditions");
```

Comment:
if this is due to coding mistakes, how about assert?

Reply:
it seems to be due to an illegal input.